public/app/features/dashboard/containers/DashboardPage.tsx

```diff
@@ -108,6 +112,12 @@ export class DashboardPage extends PureComponent<Props, State> {
       document.title = dashboard.title + ' - Grafana';
     }
 
+    // Listen for inspect events
+    if ((inspectPanelId && !this.state.inspectPanel) || prevProps.inspectPanelId !== inspectPanelId) {
```

Comment:
I think we don't need this. Since the inspectId is in the props we can just call `dashboard.getPanelById` in the render function and decide ther if the drawer should be rendered or not. WDYT?